src/components/CompareResults/ResultsMain.tsx

```diff
@@ -1,25 +1,46 @@
-import { Suspense, useState } from 'react';
+import { Suspense, useEffect, useState } from 'react';
 
+import { SelectChangeEvent } from '@mui/material';
 import Box from '@mui/material/Box';
 import CircularProgress from '@mui/material/CircularProgress';
 import { Container } from '@mui/system';
+import { useSearchParams } from 'react-router-dom';
 import { Await, useLoaderData } from 'react-router-dom';
 import { style } from 'typestyle';
 
 import { useAppSelector } from '../../hooks/app';
 import { Colors, Spacing } from '../../styles';
 // import type { CompareResultsItem } from '../../types/state';
+import { Repository } from '../../types/state';
+import { Framework } from '../../types/types';
 import DownloadButton from './DownloadButton';
 import type { LoaderReturnValue } from './loader';
+import ResultsFrameworkDropdown from './ResultsFrameWorkDropDown';
 import ResultsTable from './ResultsTable';
 import RevisionSelect from './RevisionSelect';
 import SearchInput from './SearchInput';
 
 function ResultsMain() {
-  const { results } = useLoaderData() as LoaderReturnValue;
+  const { results, frameworkId } = useLoaderData() as LoaderReturnValue;
 
   const themeMode = useAppSelector((state) => state.theme.mode);
+  const [searchParams, setSearchParams] = useSearchParams();
   const [searchTerm, setSearchTerm] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+  const [frameworkIdVal, setFrameworkIdVal] = useState(frameworkId);
+
+  // allows user to see the loading spinner
+  // for a short time when changing frameworks
+  useEffect(() => {
+    const waitForResults = async () => {
+      await results;
+      setTimeout(() => {
+        setIsLoading(false);
+      }, 50);
+    };
+
+    void waitForResults();
+  }, [isLoading]);
```

Comment:
Not a huge fan of this change in this PR, also see below for an accessibility problem with the focus handling.

I think that if we cache the revision info like I mentioned, and keep the Await/Suspense thing _without the `defer`_ we'd properly see the spinner. We can deal with this later, but in the mean time I'd remove the custom spinner mechanism in this PR, because it adds some complexity and I'd rather remove it later when we manage to make Await/Suspense work properly.
What do you think?

Reply:
I hear what you're saying. I want to create a new branch to test caching the revisions and removing `defer` because from what I can see from the docs, `_Suspense_` was never meant for same page updates only on initial load. However, I'll be more than happy to revert to `_Suspense_` if the proposed changes above allow for same page updates. :)

Reply:
Had to import the `ResultsTable` with lazy to show the spinner:

https://react.dev/reference/react/Suspense#displaying-a-fallback-while-content-is-loading